pandas/tests/indexing/multiindex/test_multiindex.py

```diff
@@ -0,0 +1,907 @@
+        cp = df.copy()
+        cp['a'] = cp['b'].values
+        tm.assert_frame_equal(cp['a'], cp['b'])
+
+        # ---------------------------------------
+        # #1803
+        columns = MultiIndex.from_tuples([('A', '1'), ('A', '2'), ('B', '1')])
+        df = DataFrame(index=[1, 3, 5], columns=columns)
+
+        # Works, but adds a column instead of updating the two existing ones
+        df['A'] = 0.0  # Doesn't work
+        assert (df['A'].values == 0).all()
+
+        # it broadcasts
+        df['B', '1'] = [1, 2, 3]
+        df['A'] = df['B', '1']
+
+        sliced_a1 = df['A', '1']
+        sliced_a2 = df['A', '2']
+        sliced_b1 = df['B', '1']
+        tm.assert_series_equal(sliced_a1, sliced_b1, check_names=False)
+        tm.assert_series_equal(sliced_a2, sliced_b1, check_names=False)
+        assert sliced_a1.name == ('A', '1')
+        assert sliced_a2.name == ('A', '2')
+        assert sliced_b1.name == ('B', '1')
+
+    def test_getitem_tuple_plus_slice(self):
+        # GH #671
+        df = DataFrame({'a': lrange(10),
+                        'b': lrange(10),
+                        'c': np.random.randn(10),
+                        'd': np.random.randn(10)})
+
+        idf = df.set_index(['a', 'b'])
+
+        result = idf.loc[(0, 0), :]
+        expected = idf.loc[0, 0]
+        expected2 = idf.xs((0, 0))
+        with catch_warnings(record=True):
+            simplefilter("ignore", DeprecationWarning)
+            expected3 = idf.ix[0, 0]
+
+        tm.assert_series_equal(result, expected)
+        tm.assert_series_equal(result, expected2)
+        tm.assert_series_equal(result, expected3)
+
+    def test_getitem_setitem_tuple_plus_columns(
+            self, multiindex_year_month_day_dataframe_random_data):
+        # GH #1013
+        ymd = multiindex_year_month_day_dataframe_random_data
+        df = ymd[:5]
+
+        result = df.loc[(2000, 1, 6), ['A', 'B', 'C']]
+        expected = df.loc[2000, 1, 6][['A', 'B', 'C']]
+        tm.assert_series_equal(result, expected)
+
+    def test_get_loc_single_level(self, single_level_multiindex):
+        single_level = single_level_multiindex
+        s = Series(np.random.randn(len(single_level)),
+                   index=single_level)
+        for k in single_level.values:
+            s[k]
+
+    def test_getitem_toplevel(self, multiindex_dataframe_random_data):
+        frame = multiindex_dataframe_random_data
+        df = frame.T
+
+        result = df['foo']
+        expected = df.reindex(columns=df.columns[:3])
+        expected.columns = expected.columns.droplevel(0)
+        tm.assert_frame_equal(result, expected)
+
+        result = df['bar']
+        result2 = df.loc[:, 'bar']
+
+        expected = df.reindex(columns=df.columns[3:5])
+        expected.columns = expected.columns.droplevel(0)
+        tm.assert_frame_equal(result, expected)
+        tm.assert_frame_equal(result, result2)
+
+    def test_getitem_setitem_slice_integers(self):
```

Comment:
can you move pure getitem tests to a new file, and pure setitem to a newfile as well, if they are both getitem & setitem, put in the setitem one